### PR TITLE
Pol 351 azure schedule instance

### PR DESCRIPTION
**Pull request by gopinathmahakud1**
### Description

-Implemented new policy for azure schedule intance

### Issues Resolved

- New .pt, readme and changelog created.
- Policy link added in the root readme file.

### Contribution Check List

- [x] All tests pass.
- [x] New functionality includes testing.
- [x] New functionality has been documented in the README if applicable
- [x] New functionality has been documented in CHANGELOG.MD

https://governance.rightscale.com/org/1105/projects/60073/policy-incidents/5f3aaf40a1dbf80001e012a7
image

### CODE REVIEW

File: cost/azure/schedule_instance/README.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
+# Azure Schedule Instance
+
+## What it does
+
+This Policy Template allows you to schedule start and stop times for your instance, along with the option to terminate instance, update and delete schedule.
```
Comment (cdwilhelm) [inline cost/azure/schedule_instance/README.md:5]: Please include the cloud name in What it does

File: cost/azure/schedule_instance/README.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+### Credential configuration
+
+For administrators [creating and managing credentials](https://docs.rightscale.com/policies/users/guides/credential_management.html) to use with this policy, the following information is needed:
+
+Provider tag value to match this policy: `azure`
+
+Required permissions in the provider:
+
+- Microsoft.Compute/virtualMachines/read
```
Comment (cdwilhelm) [inline cost/azure/schedule_instance/README.md:62]: Is there a permission to modify the tags?

File: cost/azure/schedule_instance/azure_schedule_instance.pt [hunk 3]
```diff
@@ -0,0 +1,537 @@
...
+rs_pt_ver 20180301
+type "policy"
+short_description "This Policy Template allows you to schedule start and stop times for your instance, along with the option to terminate instance, update and delete schedule. See the [README](https://github.com/flexera/policy_templates/tree/master/cost/azure/schedule_instance/) and [docs.rightscale.com/policies](https://docs.rightscale.com/policies/) to learn more."
+long_description ""
+category "Cost"
+severity "low"
+info(
+  version: "2.0",
+  provider: "Azure",
+  service: "azure",
```
Comment (cdwilhelm) [inline cost/azure/schedule_instance/azure_schedule_instance.pt:11]: I believe service should be compute.

File: cost/azure/schedule_instance/azure_schedule_instance.pt [hunk 4]
```diff
@@ -0,0 +1,537 @@
...
+      field "state" do
+        label "State"
+      end
+      field "schedule" do
+        label "Schedule"
+      end
+      field "subscriptionId" do
+        label "SubscriptionId"
+      end
+      field "subscriptionName" do
```
Comment (cdwilhelm) [inline cost/azure/schedule_instance/azure_schedule_instance.pt:289]: subscription name should be the first column and sorted.

File: cost/azure/schedule_instance/azure_schedule_instance.pt [hunk 5]
```diff
@@ -0,0 +1,537 @@
...
+      field "id" do
+        label "Instance ID"
+      end
+      field "name" do
+        label "Instance Name"
+      end
+      field "rg" do
+        label "Resource Group"
+      end
+      field "location" do
```
Comment (cdwilhelm) [inline cost/azure/schedule_instance/azure_schedule_instance.pt:274]: the second column should probably be location,

File: cost/azure/schedule_instance/azure_schedule_instance.pt [hunk 6]
```diff
@@ -0,0 +1,537 @@
...
+    summary_template "Azure Schedule Instance list"
+    hash_include "next_start", "next_stop"
+    escalate $esc_email
+    escalate $esc_schedule_instance
+    escalate $esc_terminate_instance
+    escalate $esc_update_schedule
+    escalate $esc_delete_schedule
+    check eq(size(data),0)
+    export do
+      resource_level true
```
Comment (cdwilhelm) [inline cost/azure/schedule_instance/azure_schedule_instance.pt:264]: Please review the columns and think about how a user may want to view the details as a default view.

File: cost/azure/schedule_instance/README.md (hunk 2 above)
Comment (gopinathmahakud1) [reply cost/azure/schedule_instance/README.md:62]: Microsoft.Compute/virtualMachines/write --Creates a new virtual machine or updates an existing virtual machine

To add and update tags we are calling update VM API.

File: cost/azure/schedule_instance/azure_schedule_instance.pt (hunk 4 above)
Comment (gopinathmahakud1) [reply cost/azure/schedule_instance/azure_schedule_instance.pt:289]: Do we need to sort the subscription, since now we can sort in UI ?

Comment (cdwilhelm) [reply cost/azure/schedule_instance/azure_schedule_instance.pt:289]: Why wouldn't we want to have the subscription sorted in the default view?

Comment (gopinathmahakud1) [reply cost/azure/schedule_instance/azure_schedule_instance.pt:289]: Yes we can do that, I will update the template.

**Comment by gopinathmahakud1** [comment]
@cdwilhelm , I have addressed the comments please review the changes.
image

**Pull request merged**